prereise/gather/demanddata/transportation_electrification/generate_BEV_vehicle_profiles.py

```diff
@@ -0,0 +1,158 @@
+import os
+
+import pandas as pd
+
+from prereise.gather.const import abv2state
+from prereise.gather.demanddata.transportation_electrification import (
+    const,
+    immediate,
+    smart_charging,
+)
+from prereise.gather.demanddata.transportation_electrification.data_helper import (
+    generate_daily_weighting,
+    get_kwhmi,
+    load_rural_scaling_factor,
+    load_urbanized_scaling_factor,
+)
+
+
+def generate_bev_vehicle_profiles(
+    vehicle_trip_data_filepath,
+    charging_strategy,
+    veh_type,
+    veh_range,
+    projection_year,
+    state,
+    external_signal=None,
+    power=6.6,
+    location_strategy=2,
+    trip_strategy=1,
+):
+    """Generate Battery Electric Vehicle (BEV) profiles
+
+    :param str vehicle_trip_data_filepath: filepath of collected trip data from external sources
+        representing driving patterns
+    :param str charging_strategy: establishes whether charging happens immediately ("immediate")
+         or optimize based on external signals, i.e. smart charging ("smart")
+    :param str veh_type: vehicle category: LDV: light duty vehicle, LDT: light duty truck,
+        MDV: medium duty vehicle, HDV: heavy duty vehicle
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on
+        single charge in miles.
+    :param int projection_year: year that is being modelled/projected to, 2017, 2030, 2040,
+        2050.
+    :param str state: US state abbreviation
+    :param np.ndnarray (optional) external_signal: initial load demand (MW for each hour)
+    :param int power: (optional) charger power, EVSE kW; default value: 6.6 kW;
+    :param int location_strategy: (optional) where the vehicle can charge-1, 2, 3, 4, or 5;
+        1-home only, 2-home and work related, 3-anywhere if possibile,
+        4-home and school only, 5-home and work and school.
+        default value: 2
+    :param int trip_strategy: (optional) determine to charge after any trip (1) or only after the
+        last trip (2); default value: 1
+    :return: (*pandas.DataFrame*) -- yearly charging profiles for all urban areas and the rural area
+        in each state (MW for each hour)
```

Comment:
This function is supposed to return yearly charging profiles for all urban areas and the rural area of the specified `state`. I guess we will need to loop through all the states in the higher level function call to get profiles for each state, no?

Reply:
Yes indeed.